docs/devel/api-conventions.md

```diff
@@ -247,6 +248,14 @@ ports:
 
 This rule maintains the invariant that all JSON/YAML keys are fields in API objects. The only exceptions are pure maps in the API (currently, labels, selectors, annotations, data), as opposed to sets of subobjects.
 
+#### Primitive types
+
+* Avoid floating-point values as much as possible, and never use them in spec. Floating-point values cannot be reliably round-tripped (encoded and re-decoded) without changing, and have varying precision and representations across languages and architectures.
+* Do not use unsigned integers. Similarly, not all languages (e.g., Javascript) support unsigned integers.
+* int64 is converted to float by Javascript and some other languages, so they also need to be accepted as strings.
```

Comment:
all numbers are converted to float64 by [...], so any number which is expected to exceed that (specifically int values > 53 bits) should be serialized and accepted as strings